### PR TITLE
Proof of concept: my artists

### PR DESCRIPTION
**Pull request by hedgehog1833**
First steps on the my-artists page. I just tried to make it look more or less like the old page and this went quite well I guess. As we already talked about new styling I did not do too much in this part (as you can see), but the functionality is more or less complete. The [new lint-rule](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-unused-vars.md) is because otherwise when defining a function in a type declaration (see PaginationProps) the linter complains about variables being declared but not used. With this new rule, this is gone (but still works for 'normal' unused variables).

### CODE REVIEW

File: src/Pages/MyArtists.tsx [hunk 1]
```diff
@@ -1,26 +1,90 @@
...
+  readonly followedSince: string
+  readonly source: string
+  readonly follower: number
+}
+
+export interface BackendPagination {
+  readonly currentPage: number
+  readonly itemsPerPage: number
+  readonly totalPages: number
+}
```
Comment (DanielW1987) [inline src/Pages/MyArtists.tsx:31]: would be nice if we have our own types file somewhere. Maybe distinguish between request/response and domain types.

File: src/Components/Pagination/DefaultPagination.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  totalPages: number
+  onChange: (event: React.ChangeEvent<unknown>, page: number) => void
+}
+
+const DefaultPagination = (props: PaginationProps) => {
+  return props.totalPages > 0 ? (
+    <Pagination count={props.totalPages} showFirstButton showLastButton boundaryCount={2} onChange={props.onChange} />
+  ) : (
+    <div />
+  )
```
Comment (DanielW1987) [inline src/Components/Pagination/DefaultPagination.tsx:14]: is it really necessary to return an empty div?


Maybe something like this works?

```suggestion
  return (props.totalPages > 0 && (
    <Pagination count={props.totalPages} showFirstButton showLastButton boundaryCount={2} onChange={props.onChange} />
  )
```

File: src/Pages/MyArtists.tsx [hunk 3]
```diff
@@ -1,26 +1,90 @@
...
-      .then((response: AxiosResponse) => {
-        console.log(`success: ${JSON.stringify(response.data)}`)
+      .then((response: AxiosResponse<MyArtistsResponse>) => {
+        setMyArtistsResponse(response.data)
       })
       .catch((error: AxiosError) => {
         console.log(`error: ${JSON.stringify(error.response?.data)}`)
       })
-  }, [])
+  }, [searchParams])
```
Comment (DanielW1987) [inline src/Pages/MyArtists.tsx:49]: In the long run I do not want to mix up components with API communication (OK for now).

File: src/Pages/MyArtists.tsx [hunk 4]
```diff
@@ -1,26 +1,90 @@
...
+          ))}
+        </Grid>
+      </div>
+      <div>
+        <DefaultPagination
+          totalPages={myArtistsResponse?.pagination.totalPages == null ? 0 : myArtistsResponse?.pagination.totalPages}
+          onChange={handleOnChange}
+        />
+      </div>
+    </Container>
```
Comment (DanielW1987) [inline src/Pages/MyArtists.tsx:88]: Pages should only be a wrapper to place components. We should think in terms of smaller components to increase reuse as well. I see at least ArtistList and ArtistCard as a component.

File: src/Pages/MyArtists.tsx [hunk 5]
```diff
@@ -1,26 +1,90 @@
...
+    event.preventDefault()
+    searchParams.set('page', page.toString())
+    setSearchParams(searchParams)
+  }
 
-  return <h1>MyArtists</h1>
+  return (
+    <Container maxWidth={'lg'}>
+      <h1>MyArtists</h1>
+      <div>
```
Comment (DanielW1987) [inline src/Pages/MyArtists.tsx:60]: the empty div tags only inflate the DOM unnecessarily

File: src/Pages/MyArtists.tsx (hunk 4 above)
Comment (hedgehog1833) [reply src/Pages/MyArtists.tsx]: Sounds sensible, I extracted ArtistCard and ArtistGrid, what do you think?

Comment (DanielW1987) [reply src/Pages/MyArtists.tsx]: 🎖️

File: src/Pages/MyArtists.tsx [hunk 6]
```diff
@@ -1,26 +1,46 @@
...
 
-  return <h1>MyArtists</h1>
+  return (
+    <Container maxWidth={'lg'}>
+      <h1>MyArtists</h1>
+      <ArtistGrid artists={myArtistsResponse?.myArtists || []} />
+      <DefaultPagination
+        totalPages={myArtistsResponse?.pagination?.totalPages || 0}
+        currentPage={myArtistsResponse?.pagination?.currentPage || 0}
+        onChange={handleOnChange}
```
Comment (DanielW1987) [inline src/Pages/MyArtists.tsx:42]: may be we can think about if the pagination should be placed into the grid...

**Review by DanielW1987 — APPROVED** [review]

File: src/Pages/MyArtists.tsx (hunk 6 above)
Comment (hedgehog1833) [reply src/Pages/MyArtists.tsx:42]: Maybe, yes... I'll see how the axios hook works out and then check again, if it is nicer in the grid

**Pull request merged**